iOS Client/DevotionsController/DevotionsCoordinator.swift

```diff
@@ -25,13 +26,53 @@ class DevotionsCoordinator: NSObject, Coordinator, StoreSubscriber {
         let devotionsController = MBDevotionsViewController.instantiateFromStoryboard()
         navigationController.pushViewController(devotionsController, animated: true)
         MBStore.sharedStore.subscribe(self)
+        let store = MBStore()
+        let devotions = store.getDevotions()
+        if devotions.count == 0 {
+            store.syncDevotions { syncedDevotions, error in
+                if error != nil {
+                    MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.error))
+                } else if let newDevotions = syncedDevotions {
+                    MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.loaded(data: newDevotions)))
+                }
+            }
+        } else {
+            MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.loaded(data: devotions)))
+        }
+    }
+    
+    func promptForNotifications() {
```

Comment:
where does this get called?

Reply:
maybe it could be called from a newState method when the devotions get loaded, that way we know we can use `getDevotions` in the body of prompt and schedule. Also might be worth throwing a bool in UserDefaults called `areNotificationsScheduled` or something so we can do nothing if it's already taken care of

Reply:
whoops this used to get called in the DevotionsCoordinator start method but must have gotten lost in the shuffle. Yeah I like having a field in UserDefaults I can add that

Reply:
I'm going to keep the promptForNotifications call after we sync devotions on the first start up of the devotions coordinator. It's not going to do anything if we have already scheduled or if the user has already been prompted. Not sure exactly where it would go in newState because we wouldn't want to schedule every time we got a new state. Could add a new property on DevotionState that said whether or not we have scheduled yet. I'll leave it for now and we can refactor if necessary.

Reply:
what you did is good